bin/test.sh

```diff
@@ -21,76 +21,51 @@ check_env() {
     fi
 }
 
-fetch_commons_tools() {
-    if [ -z "${COMMONS_TOOLS_DIR}" ]; then
-        pushd ${BIN_DIR}
-        rm -rf dcos-commons-tools/ && curl https://infinity-artifacts.s3.amazonaws.com/dcos-commons-tools.tgz | tar xz
-        popd
-        COMMONS_TOOLS_DIR=${BIN_DIR}/dcos-commons-tools/
-    fi
-}
-
-notify_github() {
-    ${COMMONS_TOOLS_DIR}/github_update.py $1 test $2
-}
-
 start_cluster() {
     if [ -n "${DCOS_URL}" ]; then
         echo "Using existing cluster: $DCOS_URL"
     else
-        notify_github pending "Starting Cluster"
         echo "Launching new cluster"
-        STDOUT=$(CCM_AGENTS=5 ${COMMONS_TOOLS_DIR}/launch_ccm_cluster.py)
-        DCOS_URL=$(echo "${STDOUT}" | jq .url)
+        CCM_JSON=$(CCM_AGENTS=5 ${COMMONS_DIR}/tools/launch_ccm_cluster.py)
+        DCOS_URL=$(echo "${CCM_JSON}" | jq .url)
         if [ $? -ne 0 -o "$DCOS_URL" = "http://" ]; then
-            notify_github failure "Cluster launch failed"
             exit 1
         fi
     fi
 }
 
 configure_cli() {
-    notify_github pending "Configuring CLI"
     dcos config set core.dcos_url "${DCOS_URL}"
     dcos config set core.ssl_verify false
-    ${COMMONS_TOOLS_DIR}/dcos_login.py
+    ${COMMONS_DIR}/tools/dcos_login.py
     dcos config show
     if [ -n "${STUB_UNIVERSE_URL}" ]; then
         dcos package repo add --index=0 spark-test "${STUB_UNIVERSE_URL}"
     fi
 }
 
-setup_permissions() {
+initialize_service_account() {
     if [ "$SECURITY" = "strict" ]; then
-        # custom configuration to enable auth stuff:
-        ${COMMONS_TOOLS_DIR}/setup_permissions.sh nobody "*" # spark's default service.role
-        ${COMMONS_TOOLS_DIR}/setup_permissions.sh root hdfs-role
+        ${COMMONS_DIR}/tools/create_service_account.sh --strict
+        ${COMMONS_DIR}/tools/setup_permissions.sh service-acct root "*"
+        ${COMMONS_DIR}/tools/setup_permissions.sh service-acct root hdfs-role
     fi
 }
 
 run_tests() {
-    notify_github pending "Running Tests"
-
     pushd tests
-    if [[ ! -d env ]]; then
-        virtualenv -p python3 env
+    if [[ ! -d venv ]]; then
+        virtualenv -p python3 venv
     fi
-    source env/bin/activate
+    source venv/bin/activate
     pip install -r requirements.txt
     py.test -s test.py
-    if [ $? -ne 0 ]; then
-        notify_github failure "Tests failed"
-        exit 1
-    fi
     popd
 }
 
 check_env
-fetch_commons_tools
```

Comment:
I added the COMMONS_DIR environment variable, and I fetch the dcos-commons dir in the Jenkins PR project: https://jenkins.mesosphere.com/service/jenkins/view/Infinity/job/spark/job/spark-build/job/spark-build-PR/

I'll add it to the other builds as well once this passes.

The github project is the source of truth, so we should use it.  I see no point in the dcos-commons-tools tarball.